Automattic/wp-calypso
# Permafail E2E: fix Google Authentication spec.

Fixes https://github.com/Automattic/wp-calypso/issues/75674.


## Proposed Changes

This PR fixes the permanently failing Google Authentications spec within the Authentications suite of specs.

Key changes:
- update POMs to use accessible locators.
- codify expected URL in the test steps.
- handle the double 2FA challenge that Google is sending our test account. I have not seen this in actual user accounts before and I have structured the E2E spec to handle the scenario where Google returns to a single 2FA challenge.

## Testing Instructions

Ensure the following build configurations are passing:
  - [x] Authentication E2E

Authentication E2E suite (limited to the Google Authenticate scenario, to save on SMS costs) passed:
![image](https://github.com/Automattic/wp-calypso/assets/6549265/66a20721-f21d-4600-868f-8fc484ec7b3e)


## Pre-merge Checklist
